src/Utilities/functions.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { Skeleton } from '@redhat-cloud-services/frontend-components';
+import { getEmailSchema } from '../actions';
+import { loaderField } from './constants';
+
+export const getSchema = (app) => {
+    return !app || !app.loaded ? loaderField : app.schema;
+};
+
+export const calculatePermissions = (permissions) => {
+    return Promise.all(
+        (Array.isArray(permissions) ? permissions : [ permissions ])
```

Comment:
what about `[premissions].flat()`? We can get rid of some additional conditions.

Reply:
That's a good point! Will update it.